src/physics/em/detail/MuBremsstrahlungInteractor.i.hh

```diff
@@ -0,0 +1,199 @@
+    UniformRealDistribution<real_type> phi(0, 2 * constants::pi);
+
+    real_type cost  = this->sample_cos_theta(epsilon, rng);
+    Real3 gamma_dir = rotate(from_spherical(cost, phi(rng)), inc_direction_);
+
+    Real3 inc_direction;
+    for (int i : range(3))
+    {
+        inc_direction[i] = particle_.momentum().value() * inc_direction_[i]
+                           - epsilon * gamma_dir[i];
+    }
+    normalize_direction(&inc_direction);
+
+    // Construct interaction for change to primary (incident) particle
+    Interaction result;
+    result.action    = Action::scattered;
+    result.energy    = units::MevEnergy{particle_.energy().value() - epsilon};
+    result.direction = inc_direction;
+    result.secondaries = {secondaries, 1};
+
+    // Save outgoing secondary data
+    secondaries[0].particle_id = shared_.gamma_id;
+    secondaries[0].energy      = units::MevEnergy{epsilon};
+    secondaries[0].direction   = gamma_dir;
+
+    return result;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Sample cosine of the angle between incident and secondary particles.
+ */
+template<class Engine>
+CELER_FUNCTION real_type MuBremsstrahlungInteractor::sample_cos_theta(
+    real_type gamma_energy, Engine& rng) const
+{
+    const real_type gamma = particle_.lorentz_factor();
+    const real_type r_max
+        = gamma * constants::pi * real_type(0.5)
+          * min(real_type(1.0),
+                gamma * particle_.mass().value() / gamma_energy - 1);
+    const real_type r_max_sq = ipow<2>(r_max);
+
+    const real_type a = generate_canonical(rng) * r_max_sq / (1 + r_max_sq);
+
+    return std::cos(std::sqrt(a / (1 - a)) / gamma);
+}
+
+//---------------------------------------------------------------------------//
+
+CELER_FUNCTION real_type MuBremsstrahlungInteractor::differential_cross_section(
+    real_type gamma_energy) const
+{
+    real_type dxsection = 0;
+
+    if (gamma_energy >= particle_.energy().value())
+    {
+        return dxsection;
+    }
+
+    const int       atomic_number    = element_.atomic_number();
+    const real_type atomic_mass      = element_.atomic_mass().value();
+    const real_type sqrt_e           = std::sqrt(constants::euler);
+    const real_type inc_total_energy = particle_.mass().value()
+                                       + particle_.energy().value();
+    const real_type rel_energy_transfer = gamma_energy / inc_total_energy;
+    const real_type inc_mass_sq         = ipow<2>(particle_.mass().value());
+    const real_type delta = real_type(0.5) * inc_mass_sq * rel_energy_transfer
+                            / (inc_total_energy - gamma_energy);
+
+    real_type       d_n_prime, b, b1;
+    const real_type d_n = real_type(1.54)
+                          * std::pow(atomic_mass, real_type(0.27));
+
+    if (atomic_number == 1)
+    {
+        d_n_prime = d_n;
+        b         = real_type(202.4);
+        b1        = 446;
```

Comment:
I see this in a few other spots too, where some `real_type`, such as `b`, is explicitly casted again, while other numbers declared as `real_type` aren't, such as `b1`. Is there any reason to do so?

Reply:
I think either way is technically fine, though it would be probably better to do one or the other. Interestingly in your example the first one could be a demotion (double -> float) while the second would be promotion (int -> float)...